RT_UPDATE/download_nLx.R

```diff
@@ -0,0 +1,416 @@
+      
+    }
+    
+    rownames(nLx) <- Age
+    
+    return(nLx)
+    
+  }
+  
+  # this part goes until the end of function and deals with missing nLx
+  if(is.null(nLx)) {
+    
+    # At the very least we need a location data. 
+    # if no location data is provided we stop with error
+    if(is.null(location)) {
+      
+      stop("You need to provide a location to download the data for nLx")
+      
+    }
+    
+    # I check which wpp versions are available on the machine
+    # if no wpp is available we stop and offer to use our function to download one
+    # otherwise I use the latest available wpp on the machine (in case > 1) for the mortality data 
+    
+    # installed wpp versions
+    installed_wpp <- grep("^wpp\\d{4}$", rownames(installed.packages()), value = TRUE)
+    
+    # stop if none
+    if(length(installed_wpp) == 0) {
+      
+      stop("No wpp package installed. Please use function install_wpp to download the wpp package.")
+      
+    }
+    
+    # find the lates one
+    latest_wpp <- sort(installed_wpp, decreasing = TRUE)[1]
+    
+    # UPDATE:
+    # we basically want wpp to be at leat 2022. We need this for single year data.
+    # Since we have a dedicated wpp install function, we agreed that this is ok to do
+    if(latest_wpp < 2022) {
+      
+      stop("No single ages are availabe in wpp versions earlier that wpp2022. Please update the wpp package.")
+      
+    }
+    
+    # download mx1dt data from the latest package version
+    # this data contain nmx, age, territory, sex, year
+    data("mx1dt", package = latest_wpp)
+    
+    # if any date chosen are less then 1950 or more than wpp version + 1
+    if(any(nLxDatesIn < 1950, nLxDatesIn > (parse_number(latest_wpp) + 1))) {
+      
+      cat(paste0("Careful, choosing beyond range 1950-", parse_number(latest_wpp), ". "))
+      
+    }
+    
+    # User can provide location in either number (country_code)
+    # OR as a name e.g. Brazil
+    # if it is numeric, live as is
+    # if it is a name, find the coresponding country number
+    if(is.numeric(location)) {
+      
+      location_code <- location
+      
+    } else {
+      
+      # find location code from the provided location if location is misspelled return NA
+      location_code <- mx1dt %>%
+        filter(.data$name %in% location) %>% 
+        select("country_code") %>%
+        unique() %>%
+        as.numeric()
+      
+    }
+    # ------------------------------------------------------ #
+    # if we need message print what we are trying to do exactly
+    if(verbose) {
+      cat(
+        paste0(
+          "Downloading nLx data for ",
+          location,
+          ", years ",
+          paste(nLxDatesIn, collapse = ", "),
+          ", gender ",
+          gender
+        ),
+        sep = "\n"
+      )
+    }
+    
+    # standardize input strings to match what we expect
+    sex_code <- ifelse(tolower(gender) == "both",
```

Comment:
have a look around demotools for gender codes. i believe we like to use "t" instead of "b", but understand that if a user gives "b" they mean "t"